pkg/apis/storage/validation/validation.go

```diff
@@ -409,10 +409,21 @@ func ValidateCSIDriver(csiDriver *storage.CSIDriver) field.ErrorList {
 func ValidateCSIDriverUpdate(new, old *storage.CSIDriver) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&new.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
 
-	// Spec is read-only
-	// If this ever relaxes in the future, make sure to increment the Generation number in PrepareForUpdate
-	if !apiequality.Semantic.DeepEqual(old.Spec, new.Spec) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), new.Spec, "field is immutable"))
+	// immutable fields should not be mutated.
```

Comment:
should we also call validateCSIDriverSpec again so that we check the updated fields are valid?

Reply:
done.

Reply:
typically, we try to only validate fields we allow you to change (TokenRequests, I think). if there was an invalid field we also required to be immutable, they'd be stuck without a way to update (updating triggers immutable error, not updating triggers invalid error)

Reply:
done